app/src/main/java/com/netcetera/skopjepulse/countryCitySelector/CountryCityViewModel.kt

```diff
@@ -62,7 +62,33 @@ class CountryCityViewModel(application: Application) : AndroidViewModel(applicat
     }
     val jsonSelectedCities = gson.toJson(selectedCityItems)
     editor.putString(Constants.SELECTED_CITIES, jsonSelectedCities)
-    editor.commit()
+    editor.apply()
   }
 
+  /**
+   * Returns the list of countries and cities that are currently not
+   * present in the sharedPreferences
+   */
+  fun getSelectableCities(): List<Any>{
+    var checkableCities = ArrayList<Any>()
+
+    countryCityList.value?.forEach {
+      if(!checkedCityItems.contains(it))
+        checkableCities.add(it)
+    }
+
+    var i = 0
+    while(i < checkableCities.size-1){
+      if(checkableCities[i] is CountryItem && checkableCities[i+1] is CountryItem) {
+        checkableCities.removeAt(i--)
+      }
+      i++
+    }
+
+    if(checkableCities.last() is CountryItem) {
+      checkableCities.removeAt(checkableCities.lastIndex)
+    }
+
+    return checkableCities
+  }
```

Comment:
Find this code very strange. Why no use `Array.filter`?

Also what is going on with 
``` 
if(checkableCities[i] is CountryItem && checkableCities[i+1] is CountryItem) {
  checkableCities.removeAt(i--)
}
```
```
if(current item is CountryItem && next item is Country item) {
  remove current item
  reduce 'i' by one
}
increase 'i' by one
```
At the end 'i' is same as we began and also why we check for the next item to be country?

Reply:
Fixed: [Fix getSelectableCities](https://github.com/pulse-eco/pulse-eco-android/pull/5/commits/11dcc1204e049c5ee175e99498bcba7f2fd58480)

- **Why we check for the next item to be country?**
We do this to simply remove the current country item from the final list, because if the next element is country then there is not any selectable city for the current country. Then, showing the current country would be in vain. Eventually, in the end, when every possible city is selected, the only data we would have is a list of only countries which would not be able to be selected - the cities are the ones that could be selected. I hope this helped. 👍